bitsandbytes/backends/base.py

```diff
@@ -0,0 +1,133 @@
+from abc import ABC, abstractmethod
```

Comment:
Currently, the backend base contains too few functions or too many functions depending on the view. Currently, it does not provide abstractions for blockwise quantization, QLoRA-style double quantization, and 8-bit optimizers.

On the other hand, `transform` is a CUDA-specific function that probably does not need to be implemented by any other device.

This is definitely one thing that we need to discuss: what exact function do we abstract. We need to abstract everything that is needed by all devices and keep everything that is specific to CUDA in that particular backend.